core/arch/arm/kernel/kern.ld.S

```diff
@@ -346,8 +383,14 @@ SECTIONS
 	__init_mem_usage = __end - CFG_TEE_LOAD_ADDR;
 #endif
 	. = CFG_TEE_RAM_START + CFG_TEE_RAM_VA_SIZE;
+
 	_end_of_ram = .;
 
+#ifndef CFG_WITH_PAGER
+	. = ALIGN(SMALL_PAGE_SIZE);
```

Comment:
True, this is useless. I guess it is even useless to move this before `_end_of_ram = .;`.
I will remove it.